Source/interfaces/IDisplayInfo.h

```diff
@@ -60,6 +82,26 @@ namespace Exchange {
             virtual void Updated() = 0;
         };
 
+        struct EXTERNAL IHdrIterator : virtual public Core::IUnknown {
+            enum { ID = ID_CONNECTION_PROPERTIES_HDR };
+            virtual ~IHdrIterator() {}
+
+            enum HdrFormats : uint8_t {
+                HDR_NONE = 0,
+                HDR_10,
+                HDR_HLG,
+                HDR_DOLBYVISION,
+                HDR_TECHNICOLORPRIME,
+                HDR_10PLUS,
+                HDR_UNKNOWN
+            };
+
+            virtual bool IsValid() const = 0;
+            virtual bool Next() = 0;
+            virtual void Reset() = 0;
+            virtual HdrFormats HdrFormat() const = 0;
+        };
+
```

Comment:
@pwielders, based on your comment https://github.com/rdkcentral/Thunder/pull/214#discussion_r467864644, I have introduced iterator type to relay data over GetTvHdrCapabilities &  GetStbHdrCapabilities methods of this interface. Could you please confirm if these changes are in alignment with your expectations.

Reply:
I think there is no need to create such an interface (also from a performance perspective) it is easier (simpler and less coder) to just define:

    typedef RPC::IIteratorType<HdrFormats, ID_STRINGITERATOR> IHDRIterator; // This is the interface definition
    typedef RPC::IteratorType<IHDRIterator> HDRIterator;  // This is the interface definition and declarion.

After these two definitions, you have the interface definition and an implemntation ffor it. Just instantiate it by supplying, on creation, the following syntax:

        virtual IHDRIterator* HDRSystemsSupported() const
        {
            return (Core::Service<HDRIterator>::Create<IHDRIterator>(_hdrSystems));
        }

Where _hdrSystems is the std::list<HdrFormats> _hdrSystems;

And done :-)